File: front-end/src/app/albums/album-dto.ts
import type { ArtistDto } from '../artists/artist-dto';
import type { GenreDto } from '../genres/genre-dto';
import type { SongDto } from '../songs/song-dto';

export interface AlbumDto {
  id: number;
  artPath: string;
  artist?: ArtistDto;
  album: string;
  genres: GenreDto[];
  year: number;
  songs: SongDto[];
}

File: back-end/src/main/java/dev/davidsilva/music/album/AlbumDto.java
package dev.davidsilva.music.album;

import com.fasterxml.jackson.annotation.JsonInclude;
import dev.davidsilva.music.artist.ArtistDto;
import dev.davidsilva.music.genre.GenreDto;
import dev.davidsilva.music.song.SongDto;
import lombok.Data;

import java.util.List;

@JsonInclude(JsonInclude.Include.NON_DEFAULT)
@Data
public class AlbumDto {
    private int id;
    private ArtistDto artist;
    private String album;
    private List<GenreDto> genres;
    private int year;
    private List<SongDto> songs;
}
